From 1ac96b94f1c19b0da7a43b8224aa173c580507d1 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Thu, 24 Aug 2017 18:11:28 +0800
Subject: [PATCH 87/87] ARM: sun8i: v40: enable Ethernet on Banana Pi M2 Berry

Banana Pi M2 Berry features an Ethernet port, connected to the V40 SoC's
GMAC via a RTL8211E PHY.

Enable it in the device tree.

ALDO2 regulator drives the RGMII IO on the SoC, and should be enabled.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 arch/arm/boot/dts/sun8i-v40-bananapi-m2-berry.dts | 27 +++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/arch/arm/boot/dts/sun8i-v40-bananapi-m2-berry.dts b/arch/arm/boot/dts/sun8i-v40-bananapi-m2-berry.dts
index 8d92b7a5138a..568befb80ce0 100644
--- a/arch/arm/boot/dts/sun8i-v40-bananapi-m2-berry.dts
+++ b/arch/arm/boot/dts/sun8i-v40-bananapi-m2-berry.dts
@@ -98,6 +98,15 @@
 	status = "okay";
 };
 
+&gmac {
+	pinctrl-names = "default";
+	pinctrl-0 = <&gmac_rgmii_pins>;
+	phy-supply = <&reg_dc1sw>;
+	phy-handle = <&ext_rgmii_phy>;
+	phy-mode = "rgmii";
+	status = "okay";
+};
+
 &i2c0 {
 	status = "okay";
 
@@ -115,6 +124,13 @@
 
 #include "axp22x.dtsi"
 
+&reg_aldo2 {
+	regulator-always-on;
+	regulator-min-microvolt = <2500000>;
+	regulator-max-microvolt = <2500000>;
+	regulator-name = "vcc-pa";
+};
+
 &reg_aldo3 {
 	regulator-always-on;
 	regulator-min-microvolt = <2700000>;
@@ -122,6 +138,10 @@
 	regulator-name = "avcc";
 };
 
+&reg_dc1sw {
+	regulator-name = "vcc-phy";
+};
+
 &reg_dcdc1 {
 	regulator-always-on;
 	regulator-min-microvolt = <3000000>;
@@ -174,6 +194,13 @@
 	regulator-name = "vdd-1v2-sata";
 };
 
+&mdio {
+	ext_rgmii_phy: ethernet-phy@1 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <1>;
+	};
+};
+
 &mmc0 {
 	vmmc-supply = <&reg_dcdc1>;
 	bus-width = <4>;
-- 
2.13.5

